viewing_party/party.py

```diff
@@ -1,23 +1,161 @@
 # ------------- WAVE 1 --------------------
 
+# 1.1
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and (type(rating) == int or type(rating) == float):
+        movie_dict = {
+                "title": title,
+                "genre": genre,
+                "rating": rating
+            }
+        return movie_dict
+    else:
+        return None
+
+# 1.2
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+# 1.3
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+    
+
+# 1.4
+def watch_movie(user_data, title):
+    watchlist_list = user_data["watchlist"]
+    watched_list = user_data["watched"]
+
+    for movie in watchlist_list:
+        if movie["title"] == title:
+            watchlist_list.remove(movie)
+            watched_list.append(movie)
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+# 2.1
+def get_watched_avg_rating(user_data):
+    watched_list = user_data["watched"]
+    if len(watched_list) == 0:
+        return 0.0
+
+    average_rating = 0.0
+    for movie in watched_list:
+        movie_rating = movie["rating"]
+        average_rating += movie_rating
+    average_rating /= len(watched_list)
+    return average_rating 
+
+
+# 2.2
+def get_most_watched_genre(user_data):
+    watched_list = user_data["watched"]
+    if len(watched_list) == 0:
+        return None
+    
+    most_watched_genre = None
+    most_watched_count = 0
+    genre_count = {}
+    for watched_movie in watched_list:
+        genre = watched_movie["genre"]
+        if genre in genre_count:
+            genre_count[genre] += 1
+        else:
+            genre_count[genre] = 1
+
+        if most_watched_count < genre_count[genre]:
+            most_watched_count = genre_count[genre]
+            most_watched_genre = genre

```

Comment:
Instead of doing this check after every single genre is added, it would be better to go through the dictionary after this for loop to find the keys with the highest values from the `genre_count` dictionary.